src/player.cpp

```diff
@@ -7723,20 +7723,20 @@ int player::item_wear_cost( const item& it ) const
 {
     double mv = item_handling_cost( it );
 
-    switch( it.get_layer() ) {
-        case UNDERWEAR:
+    switch( it.layer() ) {
```

Comment:
The `mv` modification should be part of the `clothing_layer` instead of this switch (which won't handle new layers).